evap/rewards/tools.py

```diff
@@ -123,4 +124,27 @@ def grant_reward_points_after_delete(instance, action, reverse, pk_set, **_kwarg
                     grantings.append(granting)
 
         if grantings:
-            RewardPointGranting.granted_by_removal.send(sender=RewardPointGranting, grantings=grantings)
+            RewardPointGranting.granted_by_participation_removal.send(sender=RewardPointGranting, grantings=grantings)
+
+
+@receiver(models.signals.pre_delete, sender=Evaluation)
+def grant_reward_points_on_evaluation_delete(instance, **_kwargs):
+    if not inside_transaction():
+        # This will always be true in a django TestCase, so our tests can't meaningfully catch calls that are not
+        # wrapped in a transaction. Requiring a transaction is a precaution so that an (unlikely) failing .delete()
+        # execution afterwards doesn't leave us in half-deleted state. Chances are, if deletion fails, staff users
+        # will still want to delete the instance.
+        # Currently, only staff:evaluation_delete and staff:semester_delete call .delete()
+        logger.exception("Called while not inside transaction")
```

Comment:
@niklasmohrin this is all a bit annoying. Ideally, I'd like to
1. start a transaction
2. store a list of participants
3. delete the evaluation, including all m2m-participants
4. create RewardPointGrantings (with transaction rollback on failure)

While the signal model allows me to run code before and after deletion, I don't see a nice way to pass the list of participants to it without having global state

I think this is fine for now. If we ever run into a "wanted to delete, failed, now need cleanup"-state, we will likely need manual cleanup anyway.

I'll look into whether we can ditch the signals and use a custom manager instead, which is what the [Django docs](https://docs.djangoproject.com/en/5.1/ref/signals/) propose to do instead:
> Signals can make your code harder to maintain. Consider implementing a helper method on a [custom manager](https://docs.djangoproject.com/en/5.1/topics/db/managers/#custom-managers), to both update your models and perform additional logic, or else [overriding model methods](https://docs.djangoproject.com/en/5.1/topics/db/models/#overriding-model-methods) before using model signals.